### PR TITLE
[PTX-1641] Added AKS node driver

### PR DESCRIPTION
**Pull request by vinayakshnd**
Depends on PR in cloudops: https://github.com/libopenstorage/cloudops/pull/45

### CODE REVIEW

File: drivers/node/aks/aks.go [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package aks
+
+import (
+	"os"
+	"time"
+
+	"github.com/sirupsen/logrus"
+
```
Comment (harsh-px) [inline drivers/node/aks/aks.go]: remove extra line

Comment (vinayakshnd) [reply drivers/node/aks/aks.go]: Done

File: drivers/node/aks/aks.go [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+/*func (g *gke) DeleteNode(node node.Node, timeout time.Duration) error {
+
+	err := a.ops.DeleteInstance(node.Name, node.Zone, timeout)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+*/
```
[the remark this answers is not in the document]
Comment (vinayakshnd) [reply drivers/node/aks/aks.go]: Done

**Comment by vinayakshnd** [comment]
Jenkins Job with code changes:
https://jenkins.portworx.co/job/Torpedo/job/dummy-tp-2-3-aks/29/console

**Comment by rodrigc** [comment]
Torpedo has been converted to go modules.
Do add new vendor dependencies you need to follow instructions similar to:

https://github.com/portworx/spawn/blob/master/docs/go-modules.md

**Comment by vinayakshnd** [comment]
Performed bellow steps locally after rebasing the branch with master (after removing Gopkg.lock)
```
Vinayaks-MacBook-Pro:torpedo vinayak$ go get github.com/libopenstorage/cloudops@master
go: finding github.com/libopenstorage/cloudops master

Vinayaks-MacBook-Pro:torpedo vinayak$ make vendor-update
go mod download
go: finding github.com/prometheus/prometheus v2.9.2+incompatible
go: finding github.com/prometheus/prometheus v2.9.2+incompatible
github.com/prometheus/prometheus@v2.9.2+incompatible: invalid version: +incompatible suffix not allowed: module contains a go.mod file, so semantic import versioning is required
make: *** [vendor-update] Error 1
Vinayaks-MacBook-Pro:torpedo vinayak$ make vendor
go mod vendor
```

```
Vinayaks-MacBook-Pro:torpedo vinayak$ make build
mkdir -p /Users/vinayak/go/src/github.com/portworx/torpedo/bin
go build -tags "daemon"  github.com/portworx/torpedo/drivers github.com/portworx/torpedo/drivers/node github.com/portworx/torpedo/drivers/node/aks github.com/portworx/torpedo/drivers/node/aws github.com/portworx/torpedo/drivers/node/gke github.com/portworx/torpedo/drivers/node/ssh github.com/portworx/torpedo/drivers/scheduler github.com/portworx/torpedo/drivers/scheduler/dcos github.com/portworx/torpedo/drivers/scheduler/k8s github.com/portworx/torpedo/drivers/scheduler/openshift github.com/portworx/torpedo/drivers/scheduler/spec github.com/portworx/torpedo/drivers/volume github.com/portworx/torpedo/drivers/volume/portworx github.com/portworx/torpedo/drivers/volume/portworx/schedops github.com/portworx/torpedo/pkg/errors github.com/portworx/torpedo/pkg/log
# github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/agentpools.go:116:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/agentpools.go:201:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/agentpools.go:291:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/agentpools.go:376:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:136:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:216:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:300:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:374:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:489:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:606:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/containerservice/mgmt/2019-02-01/containerservice/containerservices.go:606:8: too many errors
# github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:106:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:183:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:259:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:335:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:450:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:528:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/availabilitysets.go:645:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/containerservices.go:133:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/containerservices.go:213:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/containerservices.go:297:8: undefined: autorest.GetSendDecorators
vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-06-01/compute/containerservices.go:297:8: too many errors
make: *** [build] Error 2
```

**Comment by nikolaypopov** [comment]
What go version are you using? I had similar issues with using go version below 1.13.

**Comment by harsh-px** [comment]
@vinayakshnd this issue is discussed in https://github.com/golang/go/issues/33558

Use the replace directive to workaround this.

**Comment by thiguetta** [comment]
@vinayakshnd I guess it's better collect everything you did, close this PR and start a new one from fresh master

**Comment by rodrigc** [comment]
I'm shifting gears today to focus on Torpedo reviews, so will help you with this @vinayakshnd

**Comment by rodrigc** [comment]
Moved to: https://github.com/portworx/torpedo/pull/398

**Pull request closed without merging**